mdstat.c

```diff
@@ -110,6 +110,28 @@ static int add_member_devname(struct dev_member **m, char *name)
 	return 1;
 }
 
+/* Detach element from the list, it may modify list_head */
+static void mdstat_ent_list_detach_element(struct mdstat_ent **list_head, struct mdstat_ent *el)
+{
+	struct mdstat_ent *ent = *list_head;
+
+	if (ent == el) {
+		*list_head = ent->next;
+		ent->next = NULL;
+		return;
+	}
+
+	while (ent) {
```

Comment:
This is just a thought, but maybe we could use "do while" here to get rid of this first check.

Reply:
Reworked